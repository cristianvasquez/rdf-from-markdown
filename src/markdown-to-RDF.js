import { simpleAst } from 'docs-and-graphs'
import { astTriplifier } from './triplifiers/astTriplifier.js'

const defaultOptions = {
  splitOnTag: false,
  splitOnHeader: false,
  splitOnId: true,
  addLabels: false,
  includeWikipaths: true,
}

function markdownToRDF (fullText, { termMapper, pointer, path }, options = {}) {

  const json = simpleAst(fullText, { normalize: true, inlineAsArray: true })
  return astTriplifier(json, {
    pointer, termMapper, path,
  }, { ...defaultOptions, ...options })

}

export { markdownToRDF }
