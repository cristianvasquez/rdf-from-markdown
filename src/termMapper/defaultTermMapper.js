import rdf from '../rdf-ext.js'
import { SimpleUriMinter } from './uriMinter.js'

function createTermMapper ({
  baseNamespace, customMapper, getPathByName,
}) {

  const pathUriMinter = new SimpleUriMinter(`${baseNamespace().value}note/`)
  const propertyUriMinter = new SimpleUriMinter(`${baseNamespace().value}property/`)

  const maybeMapped = (txt) => customMapper ? customMapper(txt) : undefined

  // returns: NamedNode
  function pathToUri (path) {
    return pathUriMinter.toUri(path)
  }

  // To build properties
  // For example, "has name" -> http://some-vault/has-name
  function newProperty (txt, options) {
    return propertyUriMinter.toUri(txt)
  }

  function newLiteral (txt, options) {
    return rdf.literal(txt)
  }

  // http://example.com/ + ^blockId -> http://example.com/blockId
  const blockUri = (uri, blockId) => rdf.namedNode(
    `${uri.value}/${blockId.replace(/^\^/, '')}`)

  return {
    newLiteral,
    newProperty,
    maybeMapped,
    blockUri,
    pathToUri,
    pathUriMinter,
    propertyUriMinter,
    getPathByName: getPathByName ? getPathByName.path : (x) => `${x}.md`,
  }

}

export {
  createTermMapper,
}
