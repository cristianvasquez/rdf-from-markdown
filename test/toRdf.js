import { expect } from 'expect'
import toMatchSnapshot from 'expect-mocha-snapshot'
import grapoi from 'grapoi'
import rdf from 'rdf-ext'
import { markdownToRDF } from '../src/markdown-to-RDF.js'
import ns from '../src/namespaces.js'

import { createTermMapper } from '../src/termMapper/defaultTermMapper.js'
import { prettyPrint } from './support/serialization.js'

import {
  allTests,
  splitOnHeaders,
  splitOnIdentifiers,
  splitOnTags,
} from './tests.js'

expect.extend({ toMatchSnapshot })
const baseNamespace = rdf.namespace('http://my-vault.org/')

const context = {
  termMapper: createTermMapper({

    documentUri: ns.ex.document,
  }),
  path: 'file.md',
  pointer: grapoi({ dataset: rdf.dataset(), factory: rdf, term: ns.ex.file }),
}

describe('toRDF', async function () {
  for (const current of allTests) {
    it(current.title, async function () {
      const fullText = current.markdown
      const pointer = markdownToRDF(fullText, context, { baseNamespace })
      const pretty = await prettyPrint(pointer.dataset)
      expect(pretty).toMatchSnapshot(this)
    })
  }
})

describe('splitOnTag', async function () {
  it('splitOnTag:false', async function () {
    const fullText = splitOnTags.markdown
    const options = { splitOnTag: false, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })

  it('splitOnTag:true', async function () {
    const fullText = splitOnTags.markdown
    const options = { splitOnTag: true, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })

  it('splitOnTag:true addLabels:true', async function () {
    const fullText = splitOnTags.markdown
    const options = { splitOnTag: true, addLabels: true, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })
})

describe('splitOnId', async function () {
  it('splitOnId:false', async function () {
    const fullText = splitOnIdentifiers.markdown
    const options = { splitOnId: false, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })

  it('splitOnId:true', async function () {
    const fullText = splitOnIdentifiers.markdown
    const options = { splitOnId: true, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })

  it('splitOnId:true addLabels:true', async function () {
    const fullText = splitOnIdentifiers.markdown
    const options = { splitOnId: true, addLabels: true, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })
})

describe('splitOnHeader', async function () {
  it('splitOnHeader:false', async function () {
    const fullText = splitOnHeaders.markdown
    const options = { splitOnHeader: false, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })

  it('splitOnHeader:true', async function () {
    const fullText = splitOnHeaders.markdown
    const options = { splitOnHeader: true, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })

  it('splitOnHeader:true addLabels:true', async function () {
    const fullText = splitOnHeaders.markdown
    const options = { splitOnHeader: true, addLabels: true, baseNamespace }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })

  it('splitOnHeader:true includeWikiPaths:true', async function () {
    const fullText = splitOnHeaders.markdown
    const options = {
      splitOnHeader: true,
      includeWikiPaths: true,
      baseNamespace,
    }
    const pointer = markdownToRDF(fullText, context, options)
    const pretty = await prettyPrint(pointer.dataset)
    expect(pretty).toMatchSnapshot(this)
  })

})
