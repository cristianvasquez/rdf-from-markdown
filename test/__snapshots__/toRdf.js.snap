// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splitOnHeader splitOnHeader:false 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	<http://my-vault.org/property/YWdl> \\"37\\", \\"42\\" ;
	<http://my-vault.org/property/bmFtZQ==> \\"Alice\\", \\"Bob\\", \\"Charlie\\" .

"
`;

exports[`splitOnHeader splitOnHeader:true 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	dot:contains [
		dot:wikiPath \\"file.md#People\\" ;
		dot:contains [
			dot:wikiPath \\"file.md#Alice\\" ;
			<http://my-vault.org/property/YWdl> \\"37\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Alice\\" ;
		], [
			dot:wikiPath \\"file.md#Bob\\" ;
			<http://my-vault.org/property/YWdl> \\"42\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Bob\\" ;
		], [
			dot:wikiPath \\"file.md#Charlie\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Charlie\\" ;
		] ;
	] .

"
`;

exports[`splitOnHeader splitOnHeader:true addLabels:true 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	schema:name \\"file\\" ;
	dot:contains [
		dot:wikiPath \\"file.md#People\\" ;
		schema:name \\"People\\" ;
		dot:contains [
			dot:wikiPath \\"file.md#Alice\\" ;
			schema:name \\"Alice\\" ;
			<http://my-vault.org/property/YWdl> \\"37\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Alice\\" ;
		], [
			dot:wikiPath \\"file.md#Bob\\" ;
			schema:name \\"Bob\\" ;
			<http://my-vault.org/property/YWdl> \\"42\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Bob\\" ;
		], [
			dot:wikiPath \\"file.md#Charlie\\" ;
			schema:name \\"Charlie\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Charlie\\" ;
		] ;
	] .

<http://my-vault.org/property/YWdl> schema:name \\"age\\" .

<http://my-vault.org/property/bmFtZQ==> schema:name \\"name\\" .

"
`;

exports[`splitOnHeader splitOnHeader:true includeWikiPaths:true 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	dot:contains [
		dot:wikiPath \\"file.md#People\\" ;
		dot:contains [
			dot:wikiPath \\"file.md#Alice\\" ;
			<http://my-vault.org/property/YWdl> \\"37\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Alice\\" ;
		], [
			dot:wikiPath \\"file.md#Bob\\" ;
			<http://my-vault.org/property/YWdl> \\"42\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Bob\\" ;
		], [
			dot:wikiPath \\"file.md#Charlie\\" ;
			<http://my-vault.org/property/bmFtZQ==> \\"Charlie\\" ;
		] ;
	] .

"
`;

exports[`splitOnId splitOnId:false 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	<http://my-vault.org/property/Zm9hZjprbm93cw==> <http://my-vault.org/note/QWxpc29u>, <http://my-vault.org/note/QWxpY2U=> .

<http://my-vault.org/note/QWxpc29u> dot:wikiPath \\"Alison\\" .

<http://my-vault.org/note/QWxpY2U=> dot:wikiPath \\"Alice\\" .

"
`;

exports[`splitOnId splitOnId:true 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	dot:contains <http://my-vault.org/note/ZmlsZS5tZA==/alice>, <http://my-vault.org/note/ZmlsZS5tZA==/alison> .

<http://my-vault.org/note/ZmlsZS5tZA==/alice> dot:wikiPath \\"file.md#Alice\\" ;
	<http://my-vault.org/property/Zm9hZjprbm93cw==> <http://my-vault.org/note/QWxpc29u> .

<http://my-vault.org/note/QWxpc29u> dot:wikiPath \\"Alison\\" .

<http://my-vault.org/note/ZmlsZS5tZA==/alison> dot:wikiPath \\"file.md#Alison\\" ;
	<http://my-vault.org/property/Zm9hZjprbm93cw==> <http://my-vault.org/note/QWxpY2U=> .

<http://my-vault.org/note/QWxpY2U=> dot:wikiPath \\"Alice\\" .

"
`;

exports[`splitOnId splitOnId:true addLabels:true 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	schema:name \\"file\\" ;
	dot:contains <http://my-vault.org/note/ZmlsZS5tZA==/alice>, <http://my-vault.org/note/ZmlsZS5tZA==/alison> .

<http://my-vault.org/note/ZmlsZS5tZA==/alice> dot:wikiPath \\"file.md#Alice\\" ;
	schema:name \\"Alice\\" ;
	<http://my-vault.org/property/Zm9hZjprbm93cw==> <http://my-vault.org/note/QWxpc29u> .

<http://my-vault.org/note/QWxpc29u> dot:wikiPath \\"Alison\\" ;
	schema:name \\"Alison\\" .

<http://my-vault.org/property/Zm9hZjprbm93cw==> schema:name \\"foaf:knows\\" .

<http://my-vault.org/note/ZmlsZS5tZA==/alison> dot:wikiPath \\"file.md#Alison\\" ;
	schema:name \\"Alison\\" ;
	<http://my-vault.org/property/Zm9hZjprbm93cw==> <http://my-vault.org/note/QWxpY2U=> .

<http://my-vault.org/note/QWxpY2U=> dot:wikiPath \\"Alice\\" ;
	schema:name \\"Alice\\" .

"
`;

exports[`splitOnTag splitOnTag:false 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	dot:tag \\"person\\" ;
	<http://my-vault.org/property/YWdl> \\"37\\", \\"42\\" ;
	<http://my-vault.org/property/bmFtZQ==> \\"Alice\\", \\"Bob\\", \\"Charlie\\" .

"
`;

exports[`splitOnTag splitOnTag:true 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	dot:contains [
		dot:wikiPath \\"file.md#Alice\\" ;
		dot:tag \\"person\\" ;
		<http://my-vault.org/property/YWdl> \\"37\\" ;
		<http://my-vault.org/property/bmFtZQ==> \\"Alice\\" ;
	], [
		dot:wikiPath \\"file.md#Bob\\" ;
		dot:tag \\"person\\" ;
		<http://my-vault.org/property/YWdl> \\"42\\" ;
		<http://my-vault.org/property/bmFtZQ==> \\"Bob\\" ;
	] ;
	<http://my-vault.org/property/bmFtZQ==> \\"Charlie\\" .

"
`;

exports[`splitOnTag splitOnTag:true addLabels:true 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	schema:name \\"file\\" ;
	dot:contains [
		dot:wikiPath \\"file.md#Alice\\" ;
		schema:name \\"Alice\\" ;
		dot:tag \\"person\\" ;
		<http://my-vault.org/property/YWdl> \\"37\\" ;
		<http://my-vault.org/property/bmFtZQ==> \\"Alice\\" ;
	], [
		dot:wikiPath \\"file.md#Bob\\" ;
		schema:name \\"Bob\\" ;
		dot:tag \\"person\\" ;
		<http://my-vault.org/property/YWdl> \\"42\\" ;
		<http://my-vault.org/property/bmFtZQ==> \\"Bob\\" ;
	] ;
	<http://my-vault.org/property/bmFtZQ==> \\"Charlie\\" .

<http://my-vault.org/property/YWdl> schema:name \\"age\\" .

<http://my-vault.org/property/bmFtZQ==> schema:name \\"name\\" .

"
`;

exports[`toRDF BlockIds 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	dot:contains <http://my-vault.org/note/ZmlsZS5tZA==/1>, <http://my-vault.org/note/ZmlsZS5tZA==/2>, <http://my-vault.org/note/ZmlsZS5tZA==/3> ;
	<http://my-vault.org/property/ZGVzY3JpcHRpb24=> \\"Section 4\\" ;
	dot:tag \\"Section\\" .

<http://my-vault.org/note/ZmlsZS5tZA==/1> dot:wikiPath \\"file.md#Section\\" ;
	<http://my-vault.org/property/ZGVzY3JpcHRpb24=> \\"Section 1\\" ;
	dot:related <http://my-vault.org/note/Qm9iIERldGFpbHMjXjI=> .

<http://my-vault.org/note/ZmlsZS5tZA==/2> dot:wikiPath \\"file.md#Section\\" ;
	<http://my-vault.org/property/ZGVzY3JpcHRpb24=> \\"Section 2\\" ;
	dot:related <http://my-vault.org/note/Qm9iIERldGFpbHMjXjMg> .

<http://my-vault.org/note/ZmlsZS5tZA==/3> dot:wikiPath \\"file.md#Section\\" ;
	<http://my-vault.org/property/ZGVzY3JpcHRpb24=> \\"Section 3\\" ;
	dot:related <http://my-vault.org/note/VW5rbm93biNeMQ==> .

"
`;

exports[`toRDF Inline fields 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	<http://my-vault.org/property/aW5saW5l> \\"field\\" ;
	<http://my-vault.org/property/KipUaG91Z2h0cyoq> \\"It was decent.\\" ;
	<http://my-vault.org/property/KipSYXRpbmcqKg==> \\"6'\\" .

<http://my-vault.org/property/c3ViamVjdA==> <http://my-vault.org/property/aW5saW5l> \\"field\\" ;
	<http://my-vault.org/property/KipUaG91Z2h0cyoq> \\"It was decent.\\" ;
	<http://my-vault.org/property/KipSYXRpbmcqKg==> \\"6\\" .

<http://my-vault.org/property/VG9v> <http://my-vault.org/property/bWFueQ==> \\"inline\\" .

<http://my-vault.org/note/Qm9i> dot:wikiPath \\"Bob\\" ;
	<http://my-vault.org/property/Zm9hZjprbm93cw==> <http://my-vault.org/note/QWxpY2U=> .

<http://my-vault.org/note/QWxpY2U=> dot:wikiPath \\"Alice\\" .

"
`;

exports[`toRDF Links 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	<http://my-vault.org/property/aGFzIGltYWdl> <http://my-vault.org/note/aG91c2VzL2ltZy5wbmc=> ;
	dot:related <http://my-vault.org/note/bGluayAx>, <http://my-vault.org/note/L2Z1bGwvcGF0aC9saW5rIDE=>, <http://my-vault.org/note/L2Z1bGwvcGF0aC9saW5rIDEubWQ=>, <http://example.com>, <http://my-vault.org/note/bGluayAxIA==>, <http://my-vault.org/note/ZmlsZS5tZC9wcm90b2NvbA==>, <https://example2.com> .

<http://my-vault.org/note/aG91c2VzL2ltZy5wbmc=> dot:wikiPath \\"houses/img.png\\" .

"
`;

exports[`toRDF Outline 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" .

"
`;

exports[`toRDF People 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	<http://my-vault.org/property/bnVtYmVy> \\"42\\" ;
	<http://my-vault.org/property/YWdl> \\"42\\", \\"56\\" .

"
`;

exports[`toRDF YAML-like 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	dot:related <https://example-2.com/something>, <https://example-2.com/> .

"
`;

exports[`toRDF Yaml 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" ;
	dot:tag \\"cats\\", \\"dogs\\" ;
	<http://my-vault.org/property/ZGF0ZSB1cGRhdGVk> \\"2021-08-05T17:10:34+02:00\\" ;
	<http://my-vault.org/property/b25l> \\"Two\\", [
		<http://my-vault.org/property/MA==> \\"Two\\" ;
		<http://my-vault.org/property/MQ==> [
			<http://my-vault.org/property/VGhyZWU=> \\"Other\\" ;
		] ;
	] .

"
`;

exports[`toRDF code fence 1`] = `
"@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix ex: <http://example.org/> .
@prefix dot: <http://dottriples.org/> .
@prefix mark: <http://markdown.org/> .

<http://my-vault.org/note/ZmlsZS5tZA==> dot:wikiPath \\"file.md\\" .

"
`;
